wireapp/kalium
# fix: sending read receipts from kalium does not work on other platforms

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

sending read receipts from kalium does not work on other platforms
receive a message → open conversation with AR → other clients does not update the conversation as read

### Causes (Optional)

TL, DR every read receipt we send is for selfconversation

`UpdateConversationReadDateUseCase` is setting self conversation as both the conversation to send the read receipt to and as the conversation ID that is updated as read

### Solutions

set the conversation IDs correctly in `UpdateConversationReadDateUseCase`
cache self conversation id, so there is no need to fetch it from database every time

Needs releases with:


### Testing

#### Test Coverage (Optional)


#### How to Test

receive a message → open conversation with AR → other clients should update the conversation as read

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
